src/main/java/org/openjdk/jextract/impl/TypeImpl.java

```diff
@@ -373,7 +373,7 @@ public String toString() {
     public static Optional<MemoryLayout> getLayout(org.openjdk.jextract.Type t) {
         try {
             return Optional.of(getLayoutInternal(t));
-        } catch (Throwable ex) {
+        } catch (UnsupportedOperationException ex) {
```

Comment:
These changes are not strictly necessary, but I don't like that we swallow important exceptions and return empty optionals instead.